ood-portal-generator/lib/ood_portal_generator/application.rb

```diff
@@ -29,56 +33,241 @@ def context
         config.file? ? YAML.load_file(config) : {}
       end
 
-      # Starts the OodPortalGenerator CLI
-      # @return [void]
-      def start
-        # Set a cleaner process title
-        Process.setproctitle("generate #{ARGV.join(" ")}")
+      def rpm
+        @rpm.nil? ? false : @rpm
+      end
 
-        # Parse CLI arguments
-        parser.parse!
+      def force
+        @force.nil? ? false : @force
+      end
 
-        # Render Apache portal config
+      def detailed_exitcodes
+        @detailed_exitcodes.nil? ? false : @detailed_exitcodes
+      end
+
+      def change_exit
+        detailed_exitcodes ? 3 : 0
+      end
+
+      def skip_exit
+        detailed_exitcodes ? 4 : 0
+      end
+
+      def apache
+        ENV['APACHE'] || (OodPortalGenerator.scl_apache? ? '/opt/rh/httpd24/root/etc/httpd/conf.d/ood-portal.conf' : '/etc/httpd/conf.d/ood-portal.conf')
+      end
+
+      def apache_bak
+        "#{apache}.#{Time.now.strftime('%Y%m%dT%H%M%S')}"
+      end
+
+      def apache_services
+        if OodPortalGenerator.scl_apache?
+          ['httpd24-httpd', 'httpd24-htcacheclean']
+        else
+          ['httpd', 'htcacheclean']
+        end
+      end
+
+      def sum_path
+        ENV['SUM'] || '/etc/ood/config/ood_portal.sha256sum'
+      end
+
+      # return string contents of file without comments
+      def read_file_omitting_comments(input)
+        return '' unless File.exist?(input)
+        File.readlines(input).reject {|line| line =~ /^\s*#/ }.join('')
+      end
+
+      def checksum(input)
+        Digest::SHA256.hexdigest(read_file_omitting_comments(input))
+      end
+
+      def save_checksum(input)
+        File.write(sum_path, "#{checksum(input)} #{apache}\n")
+      end
+
+      def checksum_matches?(input)
+        checksum_str = File.readlines(sum_path)[0]
+        checksum = checksum_str.split(' ')[0]
+
+        str = read_file_omitting_comments(input)
+        new_checksum = checksum(input)
+
+        checksum == new_checksum
+      end
+
+      def checksum_exists?
+        return false unless File.exist?(sum_path)
+        File.readlines(sum_path).grep(apache).size == 0
+      end
+
+      def update_replace?
+        # If checksum of ood-portal.conf matches or ood-portal.conf doesn't exit, replace is possible.
+        # If the checksum matches this means a site has not changed ood-portal.conf outside ood-portal-generator
+        if ! File.exist?(apache)
+          replace = true
+        elsif checksum_matches?(apache)
+          replace = true
+        else
+          replace = false
+        end
+        replace = true if force
+        return replace
```

Comment:
This might be clearer by just having force be the first option:

```ruby
def update_replace?
  force || (! File.exist?(apache)) || checksum_matches?(apache)
end
```

Also the return value doesn't need to be explicitly set to `true` or `false`.

Reply:
This change was made with tests I wrote with old method and tests still passed with new method.